src/In.ProjectEKA.DefaultHip/DataFlow/Collect.cs

```diff
@@ -25,12 +27,18 @@ public Collect(string careContextMapFile)
 
         public async Task<Option<Entries>> CollectData(DataRequest dataRequest)
         {
-            var bundles = new List<Bundle>();
-            var results = FindPatientData(dataRequest);
-            foreach (var item in results)
+            var bundles = new Dictionary<string, List<Bundle>>();
+            var patientData = FindPatientData(dataRequest);
+            var careContextReferences = patientData.Keys.ToList();
+            foreach (var careContextReference in careContextReferences)
             {
-                Log.Information($"Returning file: {item}");
-                bundles.Add(await FileReader.ReadJsonAsync<Bundle>(item));
+                var bundlesForThisCareContextReference = new List<Bundle>();
+                foreach (var result in patientData.GetOrDefault(careContextReference))
+                {
+                    Log.Information($"Returning file: {result}");
+                    bundlesForThisCareContextReference.Add(await FileReader.ReadJsonAsync<Bundle>(result));
+                }
+                bundles.Add(careContextReference,bundlesForThisCareContextReference);
```

Comment:
I don't think you need to return dictionary here. You can add CC inside bundle.

Reply:
Changed it to CareBundle Class